File: android/BertSpeak/app/src/main/java/chuckcoughlin/bert/service/NotificationActionReceiver.java
package chuckcoughlin.bert.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import chuckcoughlin.bert.R;

/**
 * Receive intents as a result of button presses in the Notification dialog
 */
public class NotificationActionReceiver extends BroadcastReceiver {
    private final static String CLSS = "NotificationActionReceiver";

    /**
     * The user has selected one of the buttons on the VoiceService notification dialog.
     * @param context the voice service
     * @param incoming intent contains the desired action
     */
    @Override
    public void onReceive(Context context, Intent incoming) {

        Log.i(CLSS,"================================================================");

        String action=incoming.getAction();
        Intent intent = new Intent(context, VoiceService.class);
        intent.setAction(action);
        Log.i(CLSS,String.format("onReceive: action = %s",action));
        context.startService(intent);

        //This is used to close the notification tray
        Intent it = new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS);
        context.sendBroadcast(it);

        if(action.equalsIgnoreCase(context.getString(R.string.notificationStop))) System.exit(0);
    }
}

File: BertSpeak/app/src/main/kotlin/chuckcoughlin/bertspeak/service/NotificationActionReceiver.kt
package chuckcoughlin.bertspeak.service

import chuckcoughlin.bertspeak.R
import android.content.*
import android.util.Log

/**
 * Receive intents as a result of button presses in the Notification dialog
 */
class NotificationActionReceiver : BroadcastReceiver() {
    /**
     * The user has selected one of the buttons on the DispatchService notification dialog.
     * @param context the voice service
     * @param incoming intent contains the desired action
     */
    override fun onReceive(context: Context, incoming: Intent) {
        Log.i(CLSS, "================================================================")
        val action = incoming.action
        val intent = Intent(context, DispatchService::class.java)
        intent.action = action
        Log.i(CLSS, String.format("onReceive: action = %s", action))
        context.startService(intent)

        //This is used to close the notification tray
        val it = Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS)
        context.sendBroadcast(it)
        assert(action != null)
        if (action.equals(
                context.getString(R.string.notificationStop),
                ignoreCase = true
            )
        ) System.exit(0)
    }

    companion object {
        private const val CLSS = "NotificationActionReceiver"
    }
}